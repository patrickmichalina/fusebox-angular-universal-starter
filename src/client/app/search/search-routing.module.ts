import { SearchComponent } from './search.component';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { MetaGuard } from '@ngx-meta/core';

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: 'search',
        component: SearchComponent,
        canActivate: [MetaGuard],
        data: {
          meta: {
            title: 'Search'
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class SearchRoutingModule { }
