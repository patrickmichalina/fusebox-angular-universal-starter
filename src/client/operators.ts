import 'rxjs/add/observable/of';
import 'rxjs/add/observable/throw';
import 'rxjs/add/observable/fromPromise';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/catch';
import 'rxjs/add/operator/switchMap';
import 'rxjs/add/operator/do';
import 'rxjs/add/operator/share';
import 'rxjs/add/operator/toPromise';

import 'rxjs/operator/finally';
import 'rxjs/operator/startWith';
import 'rxjs/operator/debounceTime';
import 'rxjs/operator/auditTime';
import 'rxjs/operator/takeUntil';