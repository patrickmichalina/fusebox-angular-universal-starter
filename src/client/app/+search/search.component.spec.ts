import { Observable } from 'rxjs/Observable';
import { ISearchService, SearchService } from './search.service';
import { SearchComponent } from './search.component';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { URLSearchParams } from '@angular/http';
import { Component } from '@angular/core';
import { SearchModule } from './search.module';
import '../../operators';

describe(SearchComponent.name, () => {
  let fixture: ComponentFixture<SearchComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SearchModule],
      declarations: [TestComponent],
      providers: [
        { provide: SearchService, useValue: new MockSearchService() }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SearchComponent);
    fixture.detectChanges();
  });

  it('should match snapshot', async(() => {
    expect(fixture).toMatchSnapshot();
  }));

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined();
  }));
});

@Component({
  selector: 'test-component',
  template: '<pm-search></pm-search>'
})
class TestComponent {}

class MockSearchService implements ISearchService {
  returnValue: any[] = [];

  search(search?: string | undefined, sort?: string | undefined, order?: string | undefined): Observable<any> {
    const params = new URLSearchParams();

    if (search) params.set('q', search);
    if (sort) params.set('sort', sort);
    if (order) params.set('order', order);

    return Observable.create((observer: any) => {
      observer.next(this.returnValue);
      observer.complete();
    });
  }
}
