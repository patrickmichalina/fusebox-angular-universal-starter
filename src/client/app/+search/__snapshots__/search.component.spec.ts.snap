// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent should match snapshot 1`] = `
<pm-search
  $count={[Function Observable]}
  $items={[Function Observable]}
  $results={[Function Observable]}
  searchService={[Function MockSearchService]}
>
  <h1>
    Search Angular Projects on Github
  </h1>
  <form
    class="form-inline"
  >
    
    
    <div
      class="form-group mx-lg-3"
    >
      
      
      <input
        class="form-control"
        placeholder="Search Angular projects"
        type="text"
      />
      
    
    </div>
    
    
    <button
      class="btn btn-primary"
    >
      Search
    </button>
    
  
  </form>
  <ul>
    
    
    
    
  
  </ul>
  
</pm-search>
`;
